control/pid_longitudinal_controller/src/pid_longitudinal_controller.cpp

```diff
@@ -959,8 +970,15 @@ double PidLongitudinalController::applyVelocityFeedback(
   const double diff_vel = (target_motion.vel - current_vel) * vel_sign;
   const bool is_under_control = m_current_operation_mode.is_autoware_control_enabled &&
                                 m_current_operation_mode.mode == OperationModeState::AUTONOMOUS;
-  const bool enable_integration =
-    (std::abs(current_vel) > m_current_vel_threshold_pid_integrate) && is_under_control;
+
+  const bool vehicle_is_moving = std::abs(current_vel) > m_current_vel_threshold_pid_integrate;
+  const double time_under_control = getTimeUnderControl();
+  const bool vehicle_is_stuck =
+    m_enable_integration_at_low_speed
+      ? !vehicle_is_moving && time_under_control > m_time_threshold_before_pid_integrate
+      : false;
+
+  const bool enable_integration = (vehicle_is_moving || vehicle_is_stuck) && is_under_control;
```

Comment:
The status of whether a vehicle is stuck should not depend on the flag: `m_enable_integration_at_low_speed`?

```suggestion
  const bool vehicle_is_stuck =
     !vehicle_is_moving && time_under_control > m_time_threshold_before_pid_integrate;

  const bool enable_integration = (vehicle_is_moving || (m_enable_integration_at_low_speed && vehicle_is_stuck)) && is_under_control;
```